treeherder/seta/jobtypes.py

```diff
@@ -0,0 +1,152 @@
+        self.jobnames = []
+
+        ManageJobPriorityTable().update_job_priority_table()
+
+        with open(get_runnable_jobs_path()) as data:
+            joblist = json.loads(data.read())['results']
+
+        for job in joblist:
+            # e.g. web-platform-tests-4
+            # e.g. Ubuntu VM 12.04 x64 mozilla-inbound opt test web-platform-tests-4 OR
+            #      desktop-test-linux64/opt-web-platform-tests-4
+            testtype, job_identifier = self._testtype_job_identifier(job)
+
+            # XXX: This has the danger of falling out of date
+            if not testtype or 'mulet' in job['platform']:
+                LOG.info('Ignoring {}'.format(job['ref_data_name']))
+                continue
+
+            # XXX: This has the danger of falling out of date
+            for i in ('dep', 'nightly', 'non-unified', 'valgrind', 'build'):
+                if testtype.find(i) != -1:
+                    LOG.info('Ignoring {}'.format(job['ref_data_name']))
+                    continue
+
+            try:
+                platform = job['platform']
+                buildtype = job['platform_option']
+                self.jobtypes.append([platform, buildtype, testtype])
+
+                # It's about get jobnames and both buildbot and taskcluster job has '?'
+                # when the job_group_symbol is unknown.
+                self.jobnames.append(self._get_jobnames(job, testtype, job_identifier))
+                job_group_symbol = job['job_group_symbol'] \
+                    if job['job_group_symbol'] != '?' else ''
+                job_type_symbol = job['job_type_symbol'] if job['job_type_symbol'] else ''
+                self.tbplnames.update({testtype: {'group': job_group_symbol,
+                                                  'code': job_type_symbol}})
+            except:
+                # XXX: shold we simply let the exception bubble up?
+                LOG.exception('We failed to initialize.')
+
+    def _testtype_job_identifier(self, job):
+        job_identifier = None
+        testtype = None
+
+        # the testtype of builbot job can been found in 'ref_data_name'
+        # like web-platform-tests-4 in "Ubuntu VM 12.04 x64 mozilla-inbound
+        #  opt test web-platform-tests-4"
+        if job['build_system_type'] == 'buildbot':
+            testtype = job['ref_data_name'].split(' ')[-1]
+            job_identifier = job['ref_data_name']
+
+        # taskcluster's testtype is a part of its 'job_type_name' like reftest-2
+        # for [TC] Linux64 reftest-2
+        elif job['build_system_type'] == 'taskcluster':
+            # The test name on taskcluster comes to a sort of combination
+            # (e.g desktop-test-linux64/debug-jittests-3) and asan job can
+            # been referenced as a opt job.
+            # NOTE: jobs like 'MacOSX64 Static Analysis Opt' will be left aside
+            # because they are not a test job.
+            if job['ref_data_name'].startswith('desktop-test') or \
```

Comment:
we should make a note here that not everything with be desktop-test or android-test, for example buildbotbridge might have a different config set.